exploration/0483-beauty-contest.md

```diff
@@ -0,0 +1,282 @@
+when foo bar {Hello {$foo} you have a {$var}}
+when * * {{$foo} hello you have a {$var}}
+
+match {$foo :function option=value} {$bar :function option=value}
+when a b {  {$foo} is {$bar}  }
+when x y {  {$foo} is {$bar}  }
+when * * {  {$foo} is {$bar}  }
+
+input $var :function option=value local $foo = {$bar :function option=value}{Hello {$var}, you have a {$foo}}
+
+match {$foo} {$bar} when foo bar {Hello {$foo} you have a {$var}} when * * {{$foo} hello you have a {$var}}
+
+match {$foo :function option=value} {$bar :function option=value}when a b {  {$foo} is {$bar}  }when x y {  {$foo} is {$bar}  }when * * {  {$foo} is {$bar}  }
+```
+
+## 1. Invert for Text Mode
+
+Consumes exterior whitespace.
+
+```
+Hello world!
+
+Hello {$user}
+
+{input $var :function option=value}
+Hello {$var}
+
+{input $var :function option=value}
+{local $foo = $bar :function option=value}
+Hello {$var}, you have a {$foo}
+
+{match {$foo} {$bar}}
+{when foo bar} Hello {$foo} you have a {$var}
+{when * *} {$foo} hello you have a {$var}
+
+{match {$foo :function option=value} {$bar :function option=value}}
+{when a b} {  {$foo} is {$bar}  }
+{when x y} {  {$foo} is {$bar}  }
+{when * *} {|  |}{$foo} is {$bar}{|  |}
+
+{input $var :function option=value}{local $foo = $bar :function option=value}Hello {$var}, you have a {$foo}
+
+{match {$foo} {$bar}}{when foo bar} Hello {$foo} you have a {$var}{when * *} {$foo} hello you have a {$var}
+
+{match {$foo :function option=value}{$bar :function option=value}}{when a b} {  {$foo} is {$bar}  }{when x y} {  {$foo} is {$bar}  }{when * *} {|  |}{$foo} is {$bar}{|  |}
+```
+
+## 2. Text First, but Always Code After Code-Mode
+
+This is @mihnita's proposal, mentioned in the 2023-10-02 call.
+Start in text mode, but stay in code mode once you get there.
+Adapted @eemeli's start-in-text-mode syntax for these examples.
+
+```
+Hello world!
+
+Hello {$user}
+
+{input $var :function option=value}
+{Hello {$var}}
+
+{input $var :function option=value}
+{local $foo = $bar :function option=value}
+{Hello {$var}, you have a {$foo}}
+
+{match {$foo} {$bar}}
+{when foo bar} {Hello {$foo} you have a {$var}}
+{when * *} {{$foo} hello you have a {$var}}
+
+{match {$foo :function option=value} {$bar :function option=value}}
+{when a b} {  {$foo} is {$bar}  }
+{when x y} {  {$foo} is {$bar}  }
+
+{input $var :function option=value}{local $foo = $bar :function option=value}{Hello {$var}, you have a {$foo}}
+
+{match {$foo} {$bar}}{when foo bar} Hello {$foo} you have a {$var}{when * *}{{$foo} hello you have a {$var}}
+
+{match {$foo :function option=value}{$bar :function option=value}}{when a b} {  {$foo} is {$bar}  }{when x y} {  {$foo} is {$bar}  }
+```
+
+## 3. Use sigils for code mode
+
+Try to redues the use of `{`/`}` to just expressions and placeholders instead of the three
+uses we have now (the other use is for patterns). This requires escaping whitespace or using
+a placeholder for it. See #486 for a discussion of whitespace options.
```

Comment:
```suggestion
Try to reduce the use of `{`/`}` to just expressions and placeholders
instead of the three uses we have now (the other use is for patterns).
This requires escaping whitespace or using a placeholder for it.
See [#487](https://github.com/unicode-org/message-format-wg/pull/487)
for a discussion of whitespace options.
```